octodns/provider/cloudflare.py

```diff
@@ -585,7 +585,7 @@ def _extra_changes(self, existing, desired, changes):
         changed_records = {c.record for c in changes}
 
         for desired_record in desired.records:
-            if desired_record not in existing.records:  # Will be created
+            if desired_record not in existing_records:  # Will be created
```

Comment:
Looks like `existing_records[desired_record]` is fetched down below so this is doing two dict lookups (not that that really matters per-wise :grin:.) Something like the following would probably clean that up.

```suggestion
            existing_record = existing_records.get(desired_record, None)
            if existing_record:  # Will be created
```